kratos/solving_strategies/schemes/residual_based_implicit_time_scheme.h

```diff
@@ -0,0 +1,544 @@
+        Condition::Pointer pCondition,
+        ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        pCondition->InitializeNonLinearIteration(rCurrentProcessInfo);
+    }
+    
+    /**
+     * @brief It initializes a non-linear iteration (for an individual element)
+     * @param pElement The element to compute
+     * @param rCurrentProcessInfo The current process info instance
+     */
+
+    void InitializeNonLinearIteration(
+        Element::Pointer pElement,
+        ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        pElement->InitializeNonLinearIteration(rCurrentProcessInfo);
+    }
+
+    /**
+     * @brief This function is designed to be called in the builder and solver to introduce
+     * @param pElement The element to compute
+     * @param LHS_Contribution The LHS matrix contribution
+     * @param RHS_Contribution The RHS vector contribution
+     * @param EquationId The ID's of the element degrees of freedom
+     * @param rCurrentProcessInfo The current process info instance
+     */
+
+    void CalculateSystemContributions(
+        Element::Pointer pElement,
+        LocalSystemMatrixType& LHS_Contribution,
+        LocalSystemVectorType& RHS_Contribution,
+        Element::EquationIdVectorType& EquationId,
+        ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        KRATOS_TRY;
+
+        const std::size_t this_thread = OpenMPUtils::ThisThread();
+
+        //pElement->InitializeNonLinearIteration(rCurrentProcessInfo);
+
+        pElement->CalculateLocalSystem(LHS_Contribution,RHS_Contribution,rCurrentProcessInfo);
+
+        pElement->EquationIdVector(EquationId,rCurrentProcessInfo);
+
+        pElement->CalculateMassMatrix(mMatrix.M[this_thread],rCurrentProcessInfo);
+
+        pElement->CalculateDampingMatrix(mMatrix.D[this_thread],rCurrentProcessInfo);
+
+        AddDynamicsToLHS(LHS_Contribution, mMatrix.D[this_thread], mMatrix.M[this_thread], rCurrentProcessInfo);
+
+        AddDynamicsToRHS(pElement, RHS_Contribution, mMatrix.D[this_thread], mMatrix.M[this_thread], rCurrentProcessInfo);
+
+        KRATOS_CATCH( "" );
+    }
+
+    /**
+     * @brief This function is designed to calculate just the RHS contribution
+     * @param pElement The element to compute
+     * @param RHS_Contribution The RHS vector contribution
+     * @param EquationId The ID's of the element degrees of freedom
+     * @param rCurrentProcessInfo The current process info instance
+     */
+
+    void Calculate_RHS_Contribution(
+        Element::Pointer pElement,
+        LocalSystemVectorType& RHS_Contribution,
+        Element::EquationIdVectorType& EquationId,
+        ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        KRATOS_TRY;
+
+        const std::size_t this_thread = OpenMPUtils::ThisThread();
+
+        // Initializing the non linear iteration for the current element
+        // pElement->InitializeNonLinearIteration(rCurrentProcessInfo);
```

Comment:
why is this commented? how it is done in the other schemes?

Reply:
In th bossak is commented...

Reply:
![imagen](https://user-images.githubusercontent.com/19991680/36254015-c98f6b1e-1249-11e8-95fc-ebf22b6c7d80.png)